opm/grid/MinpvProcessor.hpp

```diff
@@ -333,6 +357,34 @@ namespace Opm
                             kk = kk_iter;
                         }
                     }
+                    else
+                    {
+                        if (kk < dims_[2] - 1 && (actnum.empty() || actnum[c]) && pv[c] > minpvv[c] &&
+                            multz(c) != 0.0)
+                        {
+                            // Check whether there is a gap to the neighbor below whose thickness is less
+                            // than MAX_GAP. In that case we need to create an NNC if there is a gap between the two cells.
+                            int kk_below = kk + 1;
+                            int c_below = ii + dims_[0] * (jj + dims_[1] * kk_below);
+
+                            if ((actnum.empty() || actnum[c_below]) && pv[c_below] > minpvv[c_below])
+                            {
+                                // Check MAX_GAP threshold
+                                std::array<double, 8> cz = getCellZcorn(ii, jj, kk, zcorn);
+                                std::array<double, 8> cz_below = getCellZcorn(ii, jj, kk_below, zcorn);
+                                bool vertically_connected = true; // If true a connection will be there anyway -> Skip NNC
+
+                                for(int i = 0; i < 4; ++i) {
+                                    vertically_connected = vertically_connected && std::abs(cz_below[i] - cz[4+1])
```

Comment:
:flushed: Yeah, this is probably not what we should do or I even wanted to do (simple typo of `4+1` instead of `4+i` with big consequences). Seems like my (simple) tests are still lacking...

There you sit in gdb staring add the grid preprocessing and wonder about the magic in uniquepoints.c and then you have to realize (accidentally) that it is here and something embarrassing like this.

I'll create a new PR...